base/applications/rapps/gui.cpp

```diff
@@ -257,6 +259,164 @@ class CAppRichEdit:
     }
 };
 
+class CAppInfoDisplay :
+    public CUiWindow<>
```

Comment:
If you don't empty out the template arguments, but use something that is a `CWindow` or even `CWindow` itself,
you can use the ATL mechanism of creating the window, registering the class etc.
This removes the need of you toying around with the pointer, associating it with your class etc.

Reply:
See for example the `RichEdit` you use.

Reply:
> If you don't empty out the template arguments, but use something that is a `CWindow` or even `CWindow` itself,
> you can use the ATL mechanism of creating the window, registering the class etc.
> This removes the need of you toying around with the pointer, associating it with your class etc.

I'm not familiar with ATL, so at first, I create the window in the most traditional way 😅. now corrected